instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/messaging_helper.rb

```diff
@@ -49,6 +49,10 @@ def apply_sns_attributes(attributes, context, client_method)
             attributes[SemanticConventions::Trace::MESSAGING_DESTINATION_KIND] = 'topic'
             attributes[SemanticConventions::Trace::MESSAGING_DESTINATION] = queue_name(context)
           end
+
+          def params(context, param)
+            defined?(context.metadata[:original_params][param]) ? context.metadata[:original_params][param] : context.params[param]
```

Comment:
Does a `context` have a different interface depending on a specific flow? Why is that?

Reply:
`context` have a different interface depending on `aws-sdk-ruby` gem version.

### aws-sdk-ruby V2
```rb
context.params[:param]
```

### aws-sdk-ruby V3
```rb
context.metadata[:original_params][:param]
```

Reply:
Thanks for explaining that. I think we need to come up with some guidance or implementation patterns for supporting different versions of a gem. 

I wonder if we could encode this in some programmatic way. May I ask that you add a method comment that includes links to the AWS SDK in the meantime?

Reply:
@arielvalentin 
After revisiting the source code of 'aws-sdk-ruby', I found out that `context.params[param]` exists both in v2 and v3. 
Updated PR to use `context.params[param]` instead.